ask-sdk-core/src/com/amazon/ask/dispatcher/impl/DefaultRequestDispatcher.java

```diff
@@ -131,17 +130,31 @@ protected DefaultRequestDispatcher(Collection<HandlerAdapter> handlerAdapters, C
             throw new AskSdkException(message);
         }
 
-        // execute any request interceptors attached to the handler chain
-        for (RequestInterceptor requestInterceptor : handlerChain.get().getRequestInterceptors()) {
-            requestInterceptor.process(input);
-        }
+        Optional<Response> response;
+        try {
+            // execute any request interceptors attached to the handler chain
+            for (RequestInterceptor requestInterceptor : handlerChain.get().getRequestInterceptors()) {
```

Comment:
Should we run the chain's request interceptors before we throw `AskSdkException` when no `HandlerAdapter` is found?

Reply:
My intention here is that we don't even enter the context of the chain (including its interceptors) if the basic preconditions aren't met (i.e.: no compatible handler adapter).